tests/templates/test_tensornetworks/test_TTN.py

```diff
@@ -0,0 +1,316 @@
+        qml.RZ(weights[0][0], wires=wires[0])
+        qml.RZ(weights[0][1], wires=wires[1])
+        qml.RZ(weights[1][0], wires=wires[2])
+        qml.RZ(weights[1][1], wires=wires[3])
+        qml.RZ(weights[2][0], wires=wires[1])
+        qml.RZ(weights[2][1], wires=wires[3])
+
+    def circuit2_block(weights, wires):
+        SELWeights = np.array(
+            [[[weights[0], weights[1], weights[2]], [weights[0], weights[1], weights[2]]]]
+        )
+        qml.StronglyEntanglingLayers(SELWeights, wires)
+
+    def circuit2_TTN(weights, wires):
+        SELWeights1 = np.array(
+            [
+                [
+                    [weights[0][0], weights[0][1], weights[0][2]],
+                    [weights[0][0], weights[0][1], weights[0][2]],
+                ]
+            ]
+        )
+        SELWeights2 = np.array(
+            [
+                [
+                    [weights[1][0], weights[1][1], weights[1][2]],
+                    [weights[1][0], weights[1][1], weights[1][2]],
+                ]
+            ]
+        )
+        SELWeights3 = np.array(
+            [
+                [
+                    [weights[2][0], weights[2][1], weights[2][2]],
+                    [weights[2][0], weights[2][1], weights[2][2]],
+                ]
+            ]
+        )
+        qml.StronglyEntanglingLayers(SELWeights1, wires=wires[0:2])
+        qml.StronglyEntanglingLayers(SELWeights2, wires=wires[2:4])
+        qml.StronglyEntanglingLayers(SELWeights3, wires=[wires[1], wires[3]])
+
+    @pytest.mark.parametrize(
+        (
+            "block",
+            "n_params_block",
+            "wires",
+            "n_block_wires",
+            "template_weights",
+            "expected_circuit",
+        ),
+        [
+            (
+                circuit1_block,
+                2,
+                [0, 1, 2, 3],
+                2,
+                [[0.1, 0.2], [-0.2, 0.3], [0.3, 0.4]],
+                circuit1_TTN,
+            ),
+            (
+                circuit2_block,
+                3,
+                [1, 2, 3, 4],
+                2,
+                [[0.1, 0.2, 0.3], [0.2, 0.3, -0.4], [0.5, 0.2, 0.3]],
+                circuit2_TTN,
+            ),
+        ],
+    )
+    def test_output(
+        self, block, n_params_block, wires, n_block_wires, template_weights, expected_circuit
+    ):
+        """Verifies that the output of the circuits is correct."""
+        dev = qml.device("default.qubit", wires=wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.TTN(wires, n_block_wires, block, n_params_block, template_weights)
+            return qml.expval(qml.PauliZ(wires=wires[-1]))
+
+        template_result = circuit()
+
+        @qml.qnode(dev)
+        def circuit():
+            expected_circuit(template_weights, wires)
+            return qml.expval(qml.PauliZ(wires=wires[-1]))
+
+        manual_result = circuit()
+        assert np.isclose(template_result, manual_result)
```

Comment:
Question: do you guarantee that the template is trainable wrt the block params? I think it should be, but for that you'd have to add diffability tests...

Reply:
I added a differentiability test. This seems to work fine except for ragged arrays. I suppose this is like the warnings in the other comment and related to how PennyLane handles the parameters :thinking: